chapters/errors.Rmd

```diff
@@ -815,6 +881,111 @@ the default name of `collate.log`.
 2.  Modify your first solution to handle nonexistent files
     and permission problems separately.
 
+### Testing error handling {#errors-ex-error-handling}
+
+In Chapter \@ref(errors), we modified `collate.py` to handle
+different types of errors associated with reading input files.
+The relevant code appears in `main`:
```

Comment:
Now that we've moved this exercise from the testing chapter into the error handling chapter, the modification to `collate.py` that is referred to here happens in the immediate previous exercise (`#errors-ex-exceptions`). 

We therefore probably just need to say something like:

In our suggested solution to the previous exercise, the `main()` function in `collate.py` reads as follows:

```
def main(args):
    """Run the command line program."""
    log_lev = logging.DEBUG if args.verbose else logging.WARNING
    logging.basicConfig(level=log_lev, filename=args.logfile)
    word_counts = Counter()
    logging.info('Processing files...')
    for fname in args.infiles:
        try:
            logging.debug(f'Reading in {fname}...')
            if fname[-4:] != '.csv':
                msg = util.ERRORS['not_csv_suffix'].format(
                      fname=fname)
                raise OSError(msg)
            with open(fname, 'r') as reader:
                logging.debug('Computing word counts...')
                update_counts(reader, word_counts)
        except FileNotFoundError:
            msg = f'{fname} not processed: File does not exist'
            logging.warning(msg)
        except PermissionError:
            msg = f'{fname} not processed: No permission to read'
            logging.warning(msg)
        except Exception as error:
            msg = f'{fname} not processed: {error}'
            logging.warning(msg)
    util.collection_to_csv(word_counts, num=args.num) 
```

Reply:
:+1: